dotCMS/src/integration-test/java/com/dotcms/content/elasticsearch/util/ESMappingUtilHelperTest.java

```diff
@@ -0,0 +1,349 @@
+
+                {  "text_as_integer", TextField.class, DataTypes.INTEGER,
+                        new String[] {"MyTextAsInteger"},  "integer" }
+        };
+    }
+
+    /**
+     * <b>Method to test:</b> Internally, it tests the method ESMappingUtilHelper.addMappingForFieldIfNeeded<p></p>
+     * <b>Test Case:</b> Given a field or an array of fields, the right ES mapping should be set, considering exclusions in the `es-content-mapping` file<p></p>
+     * <b>Expected Results:</b> Each test case should match the `expectedResult` value, which is a string with the correct datatype in ES
+     * @param testCase
+     * @param fieldType
+     * @param type
+     * @param fields
+     * @param expectedResult
+     * @throws IOException
+     * @throws DotIndexException
+     * @throws DotSecurityException
+     * @throws DotDataException
+     */
+    @UseDataProvider
+    @Test
+    public void testAddMappingForFields(final String testCase, final Class fieldType,
+            final DataTypes type,
+            final String[] fields, final String expectedResult)
+            throws IOException, DotIndexException, DotSecurityException, DotDataException {
+
+        Logger.info(ESMappingUtilHelperTest.class,
+                String.format("Testing Add Mapping for fields defined in %s template", testCase));
+
+        final ContentType contentType = new ContentTypeDataGen().nextPersisted();
+
+        String workingIndex = null;
+        //Build the index name
+        String timestamp = String.valueOf(new Date().getTime());
+        try {
+            //Adding fields
+            for (final String field : fields) {
+                final Field newField = FieldBuilder.builder(fieldType)
+                        .name(field).variable(field).dataType(type).contentTypeId(contentType.id())
+                        .indexed(true).build();
+                fieldAPI.save(newField, user);
+            }
+
+            workingIndex = new ESIndexAPI().getNameWithClusterIDPrefix(
+                    IndexType.WORKING.getPrefix() + "_" + timestamp);
+
+            //Create a working index
+            boolean result = contentletIndexAPI.createContentIndex(workingIndex);
+            //Validate
+            assertTrue(result);
+
+            for (final String field : fields) {
+
+                if (expectedResult == null) {
+                    assertFalse(UtilMethods.isSet(esMappingAPI.getFieldMappingAsMap(workingIndex,
+                            (contentType.variable() + StringPool.PERIOD + field).toLowerCase())));
+                } else {
+                    Map<String, String> mapping = (Map<String, String>) esMappingAPI
+                            .getFieldMappingAsMap(workingIndex,
+                                    (contentType.variable() + StringPool.PERIOD + field)
+                                            .toLowerCase()).get(field.toLowerCase());
+                    assertTrue(UtilMethods.isSet(mapping.get("type")));
```

Comment:
Codacy found an issue: [The String literal "type" appears 5 times in this file; the first occurrence is on line 184](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5739222)